backend/algorithms/objects/conditions.py

```diff
@@ -85,63 +101,44 @@ def is_path_to(self, course: str) -> bool:
     def validate(self, user: User) -> tuple[bool, list[str]]:
         return (True, []) if user.has_taken_course(self.course) else (False, [self.course])
 
+    def condition_to_model(self, model: cp_model.CpModel, user: User, courses: list[Tuple[cp_model.IntVar, Course]], course_variable: cp_model.IntVar) -> list[cp_model.Constraint]:
+        condition_var = get_variable(courses, self.course)
+        return [model.Add(condition_var < course_variable)] if condition_var is not None else [model.AddBoolAnd(False)]
+
+    def condition_negation(self, model: cp_model.CpModel, user: User, courses: list[Tuple[cp_model.IntVar, Course]], course_variable: cp_model.IntVar) -> list[cp_model.Constraint]:
+        condition_var = get_variable(courses, self.course)
+        return [model.Add(condition_var >= course_variable)] if condition_var is not None else [model.AddBoolAnd(True)]
+
     def __str__(self) -> str:
         return json.dumps({
             'id': self.course
         })
 
 
-class CoreqCoursesCondition(Condition):
-    """ Condition that the student has completed the course/s in or before the current term """
-
-    def __init__(self, logic: Logic = Logic.AND):
-        # An example corequisite is [COMP1511 || COMP1521 || COMP1531]. The user
-        # must have taken one of these courses either before or in the current term
-        self.courses: list[str] = []
-        self.logic: Logic = logic
+class CoreqCourseCondition(Condition):
```

Comment:
Changed this to force CompositeCategory to handle composition, instead of having duplication here.